azext_iot/tests/deviceupdate/test_adu_account_int.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import pytest
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.tests.deviceupdate.conftest import (
+    ACCOUNT_RG,
+    generate_generic_id,
+    generate_account_id
+)
+from typing import Dict
+
+cli = EmbeddedCLI()
+
+
+@pytest.mark.adu_infrastructure(location="eastus2euap", count=2, delete=False)
+def test_account_show_delete(provisioned_accounts: Dict[str, dict]):
+    acct_names = list(provisioned_accounts["accounts"].keys())
+    for acct_name in acct_names:
+        shown_account = cli.invoke(f"iot device-update account show -n {acct_name}").as_json()
+        assert shown_account["id"] == provisioned_accounts["accounts"][acct_name]["id"]
+        shown_account_rg = cli.invoke(f"iot device-update account show -n {acct_name} -g {ACCOUNT_RG}").as_json()
+        assert shown_account_rg["id"] == provisioned_accounts["accounts"][acct_name]["id"]
+
+    cli.invoke(f"iot device-update account delete -n {acct_names[0]} -y --no-wait")
+    cli.invoke(f"iot device-update account delete -n {acct_names[1]} -g {ACCOUNT_RG} -y")
+    cli.invoke(f"iot device-update account wait -n {acct_names[0]} --deleted")
+    for acct_name in acct_names:
+        assert not cli.invoke(f"iot device-update account show -n {acct_name} -g {ACCOUNT_RG}").success()
+
+
+@pytest.mark.adu_infrastructure(location="eastus2euap", public_network_access=False, sku="Free")
+def test_account_update(provisioned_accounts: Dict[str, dict]):
+    acct_names = list(provisioned_accounts["accounts"].keys())
+    target_acct_name = acct_names[0]
+
+    updated_account = cli.invoke(
+        f"iot device-update account update -n {target_acct_name} --set tags.env='test' publicNetworkAccess='Enabled'").as_json()
+    assert updated_account["provisioningState"] == "Succeeded"
+    assert updated_account["publicNetworkAccess"] == "Enabled"
+    assert updated_account["tags"]["env"] == "test"
+
+    cli.invoke(f"iot device-update account update -n {target_acct_name} --set tags.env2='staging' --no-wait")
+    cli.invoke(f"iot device-update account wait -n {target_acct_name} -g {ACCOUNT_RG} --updated")
+    updated_account = cli.invoke(f"iot device-update account show -n {target_acct_name}").as_json()
+    assert updated_account["provisioningState"] == "Succeeded"
+    assert updated_account["tags"]["env2"] == 'staging'
+
+    # Properties outside of --set should not be changed.
+    assert updated_account["publicNetworkAccess"] == "Enabled"
+    assert updated_account["tags"]["env"] == "test"
+
+
+@pytest.mark.adu_infrastructure(location="eastus2euap", tags="a=b c=d", identity="user,system,user")
+def test_account_create_identity_mixed(provisioned_accounts: Dict[str, dict]):
+    pass
```

Comment:
TODO?

Reply:
No. This is the test. The test pattern in ADU is embracing fixtures to reduce the complexity of writing test cases. In this case there is a custom location (well everything is pointing to eastus2euap which is point-in-time), tags, and mixed identities. The provisioned_accounts test fixture will provision the requested identities and assert key state items.